src/mutate/matchUps/drawPositions/directLoser.ts

```diff
@@ -283,18 +285,76 @@ export function directLoser(params) {
+      //find the loser participant side in the loser match up
+      const loserParticipantSide = updatedLoserMatchUp.sides?.find((s) => s.participantId === loserParticipantId);
+      //set the original status code to the correct side in the loser match
+      if (loserParticipantSide?.sideNumber) {
+        //find out how many assigned participants are already in the loser match up
+        const participantsCount = updatedLoserMatchUp?.sides?.reduce((count, current) => {
+          return current?.participantId ? count + 1 : count;
+        }, 0);
+
+        //if only one participant we need to bring over the status code and
+        //set it as the only one, and assign the empty side as the winner.
+        //We also consider outcomes from a double walkover in the main draw, which
+        //will not bring over a participant but it will bring over the status code.
+        //So we make sure there is only one participant and no existing status codes, otherwise
+        //it should be set as a double walkover.
+        if (participantsCount === 1 && statusCodes.length === 0) {
+          winningSide = loserParticipantSide.sideNumber === 1 ? 2 : 1;
+          //set the original status code from the original status codes
+          //this is flawed a bit, or at least the TDesk ui, as even if there are two participants
+          //for a WO/DEFAULT, the status code is always the first element.
+          statusCodes[0] = sourceMatchUpStatusCodes[0];
+        } else {
+          //there was already a participant in the loser matchup
+          //if the loser match is not already a WO or DEFAULT
+          if (![WALKOVER, DEFAULTED].includes(loserMatchUp.matchUpStatus)) {
+            //let's set the opponent as the winner
+            winningSide = loserParticipantSide.sideNumber === 1 ? 2 : 1;
+          } else {
+            //both participants are either WO or DEFAULT
+
+            //workaround for status codes
+            const currentStatusCode = statusCodes[0];
+            //set the original status code to the correct participant in the loser match up
+            statusCodes[loserParticipantSide.sideNumber - 1] = sourceMatchUpStatusCodes[0];
+            const otherSide = loserParticipantSide.sideNumber === 1 ? 2 : 1;
+            statusCodes[otherSide - 1] = currentStatusCode;
+
+            loserMatchUpStatus = DOUBLE_WALKOVER;
+            winningSide = undefined;
+          }
+        }
+      }
 
-    if (loserMatchUp?.matchUpId && sourceMatchUpStatus) {
-      // TODO: validate that winningSide is correct...
-      const winningSide = isFeedRound ? 2 : sourceWinningSide;
       const result = setMatchUpState({
+        matchUpStatus: loserMatchUpStatus,
         matchUpId: loserMatchUp.matchUpId,
+        matchUpStatusCodes: statusCodes,
         allowChangePropagation: true,
-        propagateExitStatus: true,
+        allowSingleParticipantWO: true,
+        propagateExitStatus: false,
```

Comment:
@CourtHive I thought of splitting the flag `propagateExitStatus` in two as we were controlling two different behaviours with one flag and it was getting confusing.
Now the `propagateExitStatus` is passed at on the top call of `setMatchUpStatus` and that indicates that we want to propagate the exit status of the match we are currently scoring. Then, in `directLoser`, we are updating the consolation match for the loser player but we are not propagating its exit status again, we are just allowing the match up to be marked as a WO even if there is only one participant, hence the name `allowSingleParticipantWO`.

Feedback welcome.

Reply:
this has been updated to only use `propagateExitStatus`